Master ad7768 support

stefpopa opened the pull request:
The ad7768 is an 8-channel, 24-bit resolution ADC.
   
In standard mode, ad7768 operates as the master and streams data to the
DSP or FPGA. The ad7768 supplies the data, the data clock (DCLK) and a
falling edge framing signal (DRDY) to the slave device.
    
 In this version of the driver the default configuration is set:
    - eco power mode
    - interface configuration of DCLK = MCLK/8, with CRC disabled
    - filter configuration is set to sinc5 and decimation = x1024
    - continuous conversion mode

mirceacaprioru commented on drivers/iio/adc/ad7768.c:
```diff
@@ -0,0 +1,220 @@
+#include <linux/iio/buffer-dma.h>
+#include <linux/iio/buffer-dmaengine.h>
+
+#define AD5758_WR_FLAG_MSK(x)	(0x80 | ((x) & 0x7F))
```
AD5758 -> AD7768 ?

mirceacaprioru commented on drivers/iio/adc/ad7768.c:
```diff
@@ -0,0 +1,220 @@
+	st->d16 = cpu_to_be16((AD5758_WR_FLAG_MSK(addr) << 8));
+
+	ret = spi_sync_transfer(st->spi, t, ARRAY_SIZE(t));
+	if (ret < 0)
```
Can this be changed to if (ret) ?
spi_sync_transfer returns 0 on succes else a negative value.

stefpopa replied:
Is there something wrong with if (ret < 0) ?

commodo commented on drivers/iio/adc/ad7768.c:
```diff
@@ -0,0 +1,220 @@
+		ret = ad7768_spi_reg_read(st, reg);
+		if (ret < 0) {
+			mutex_unlock(&st->lock);
+			return ret;
```
nitpick: you could replace this with a `goto` statement ; to line 115;
you can ignore this comment, if you are adding more stuff here that would change this;

also this is a preference;  this is fine as is

commodo commented on MAINTAINERS:
```diff
@@ -740,6 +740,7 @@ L:	linux-iio@vger.kernel.org
 W:	http://ez.analog.com/community/linux-device-drivers
 S:	Supported
 F:	drivers/iio/adc/ad7768.c
+F:	arch/arm/boot/dts/zynq-zed-adv7511-ad7768.dts
```
the commit title is a bit misleading ; 
this commit adds a device-tree [example]; not necessarily the support for it ;

commodo replied:
a commit description would be needed here

stefpopa commented:
Changes in v2:
* fixed typo: `AD5758_WR_FLAG_MSK` -> `AD7768_WR_FLAG_MSK`
* used a goto statement inside the `ad7768_reg_access()` functions

commodo commented on drivers/iio/adc/Kconfig:
```diff
@@ -85,6 +85,16 @@ config AD7476
+        depends on SPI
+        help
+          Say yes here to build support for Analog Devices AD7768 SPI analog to
+	  digital converter (ADC).
```
nitpick: indentation mismatch here

commodo commented on drivers/iio/adc/ad7768.c:
```diff
@@ -0,0 +1,219 @@
+	buffer = iio_dmaengine_buffer_alloc(indio_dev->dev.parent, "rx",
+					    &dma_buffer_ops, indio_dev);
+	if (IS_ERR(buffer))
+		return PTR_ERR(buffer);
```
`regulator_disable(st->vref);`  should be called here;
as well as , there should be a `ad7768_remove` function that just calls this disable function

commodo commented on Documentation/devicetree/bindings/iio/adc/adi,ad7768.txt:
```diff
@@ -0,0 +1,22 @@
+* Analog Devices AD7768 Analog to Digital Converter (ADC)
```
a commit description would be needed here

commodo commented:
otherwise this looks good

stefpopa commented:
Changes in v3:
* fixed indentation in kconfig file.
* added short commit messages for dt-bindings and devicetree commits. 
* added a `ad7768_remove() ` function and a `error_disable_reg` label where `regulator_disable()` is called.
